### PR TITLE
Update debian12-install-guide.md

### PR DESCRIPTION
**Pull request by aradep**
Fix typo in data directory for worldserver.conf.

### CODE REVIEW

File: docs/debian12-install-guide.md [hunk 1]
```diff
@@ -147,7 +147,7 @@ make -j $(nproc) install
 ```bash
 cp -n ~/server/etc/authserver.conf.dist ~/server/etc/authserver.conf
 cp -n ~/server/etc/worldserver.conf.dist ~/server/etc/worldserver.conf
-sudo sed -i -E 's/DataDir = "."/DataDir = "\~\/server\/data"/' ~/server/etc/worldserver.conf
+sudo sed -i -E 's|^DataDir = *$|DataDir = "../data"|' ~/server/etc/worldserver.conf
```
Comment (Helias) [inline docs/debian12-install-guide.md:150]: `../data` depends on the current directory of the terminal `"\~\/server\/data"/'` does not, why did you change it?

Comment (aradep) [reply docs/debian12-install-guide.md:150]: Setting Datadir as ~/server/data was causing the mmaps to not load properly. (showed mmap:0 in .gps)

Comment (aradep) [reply docs/debian12-install-guide.md]: To add to this, `DataDir = "../data"` was working for me, but was not working for someone else also on Debian 12.  Absolute path works in both of our cases so that seems the way to go.

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
thanks for your PR :+1:
